sdk/keyvault/azure_security_keyvault_secrets/tests/secret_client.rs

```diff
@@ -267,7 +267,10 @@ async fn round_trip_secret_verify_telemetry(ctx: TestContext) -> Result<()> {
             })
         },
         ExpectedInstrumentation {
-            package_name: recording.var("CARGO_PKG_NAME", None),
+            // Don't use `recording.var` here in case the recording was made with a different package version.
+            package_name: env!("CARGO_PKG_NAME").into(),
```

Comment:
This could stay as-is. At the very least, your comment and expression don't align.